src/app/qgsgeometryvalidationservice.cpp

```diff
@@ -133,12 +133,16 @@ void QgsGeometryValidationService::onGeometryChanged( QgsVectorLayer *layer, Qgs
 
 void QgsGeometryValidationService::onFeatureDeleted( QgsVectorLayer *layer, QgsFeatureId fid )
 {
+  mLayerChecks[layer].singleFeatureCheckErrors.remove( fid );
```

Comment:
There's a few QHash lookups to `mLayerChecks[layer]` in this method -- how about retrieving the value once as a reference, and reusing throughout?

Reply:
@nyalldawson , addressed, thanks.